diff --git a/node_modules/@heasy/react-native-sunmi-printer/android/build.gradle b/node_modules/@heasy/react-native-sunmi-printer/android/build.gradle
index f76ee37..ac0b327 100644
--- a/node_modules/@heasy/react-native-sunmi-printer/android/build.gradle
+++ b/node_modules/@heasy/react-native-sunmi-printer/android/build.gradle
@@ -13,16 +13,16 @@ buildscript {
 
 apply plugin: 'com.android.library'
 
-def safeExtGet(prop, fallback) {
-  rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['SunmiPrinter_' + name]).toInteger()
 }
 
 android {
-  compileSdkVersion safeExtGet('compileSdkVersion', 29)
-  buildToolsVersion safeExtGet('buildToolsVersion', '29.0.2')
+  compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
+  buildToolsVersion getExtOrIntegerDefault('buildToolsVersion')
   defaultConfig {
-    minSdkVersion safeExtGet('minSdkVersion', 21)
-    targetSdkVersion safeExtGet('targetSdkVersion', 29)
+    minSdkVersion getExtOrIntegerDefault('minSdkVersion')
+    targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
     versionCode 1
     versionName "1.0"
 
@@ -55,5 +55,5 @@ repositories {
 dependencies {
   //noinspection GradleDynamicVersion
   implementation "com.facebook.react:react-native:+"  // From node_modules
-  implementation "com.sunmi:printerlibrary:1.0.13"
+  implementation "com.sunmi:printerlibrary:1.0.22"
 }
diff --git a/node_modules/@heasy/react-native-sunmi-printer/android/src/main/java/com/reactnativesunmiprinter/SunmiPrinterModule.java b/node_modules/@heasy/react-native-sunmi-printer/android/src/main/java/com/reactnativesunmiprinter/SunmiPrinterModule.java
index ba89482..76edfe6 100644
--- a/node_modules/@heasy/react-native-sunmi-printer/android/src/main/java/com/reactnativesunmiprinter/SunmiPrinterModule.java
+++ b/node_modules/@heasy/react-native-sunmi-printer/android/src/main/java/com/reactnativesunmiprinter/SunmiPrinterModule.java
@@ -411,6 +411,17 @@ public class SunmiPrinterModule extends ReactContextBaseJavaModule {
     printerService.printBitmapCustom(bitmap, type, innerResultCallback);
   }
 
+  @ReactMethod
+  public void printBitmapCustomBase64(String encodedString, int pixelWidth, int type) throws RemoteException {
+    final String pureBase64Encoded = encodedString.substring(encodedString.indexOf(",") + 1);
+    final byte[] decodedBytes = Base64.decode(pureBase64Encoded, Base64.DEFAULT);
+    Bitmap decodedBitmap = BitmapFactory.decodeByteArray(decodedBytes, 0, decodedBytes.length);
+	  int w = decodedBitmap.getWidth();
+    Integer h = decodedBitmap.getHeight();
+    Bitmap scaledImage = Bitmap.createScaledBitmap(decodedBitmap, pixelWidth, (pixelWidth / w) * h, false);
+    printerService.printBitmapCustom(scaledImage, type, innerResultCallback);
+  }
+
   /**
    * 打印⼀维条码
    *
diff --git a/node_modules/@heasy/react-native-sunmi-printer/lib/typescript/index.d.ts b/node_modules/@heasy/react-native-sunmi-printer/lib/typescript/index.d.ts
index 87350d8..c3c2f69 100644
--- a/node_modules/@heasy/react-native-sunmi-printer/lib/typescript/index.d.ts
+++ b/node_modules/@heasy/react-native-sunmi-printer/lib/typescript/index.d.ts
@@ -271,6 +271,9 @@ type SunmiPrinterType = {
      * @param type
      */
     printBitmapCustom: (bitmap: any, type: number) => void;
+
+    printBitmapCustomBase64: (encodedString: string, pixelWidth:number, type: number) => void;
+    
     /**
      * 是否存在打印机服务
      */
diff --git a/node_modules/@heasy/react-native-sunmi-printer/src/index.tsx b/node_modules/@heasy/react-native-sunmi-printer/src/index.tsx
index 0246c42..5e9a9c1 100644
--- a/node_modules/@heasy/react-native-sunmi-printer/src/index.tsx
+++ b/node_modules/@heasy/react-native-sunmi-printer/src/index.tsx
@@ -327,6 +327,9 @@ type SunmiPrinterType = {
    * @param type
    */
   printBitmapCustom: (bitmap: any, type: number) => void;
+
+  printBitmapCustomBase64: (encodedString: string, pixelWidth:number, type: number) => void;
+  
   /**
    * 是否存在打印机服务
    */
