diff --git a/node_modules/@makgabri/react-native-sunmi-printer/android/build.gradle b/node_modules/@makgabri/react-native-sunmi-printer/android/build.gradle
index 874c6ca..88b114b 100644
--- a/node_modules/@makgabri/react-native-sunmi-printer/android/build.gradle
+++ b/node_modules/@makgabri/react-native-sunmi-printer/android/build.gradle
@@ -14,15 +14,15 @@ buildscript {
 
 apply plugin: 'com.android.library'
 
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+def getExtOrIntegerDefault(name) {
+  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['SunmiPrinter_' + name]).toInteger()
 }
 
 android {
-    compileSdkVersion safeExtGet('SunmiPrinter_compileSdkVersion', 29)
+    compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')
     defaultConfig {
-        minSdkVersion safeExtGet('SunmiPrinter_minSdkVersion', 16)
-        targetSdkVersion safeExtGet('SunmiPrinter_targetSdkVersion', 29)
+        minSdkVersion getExtOrIntegerDefault('minSdkVersion')
+        targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
         versionCode 1
         versionName "1.0"
 
